pages/common/xml.md

```diff
@@ -0,0 +1,29 @@
+# xml
+
+> XMLStarlet Toolkit: Query, edit, check, convert and transform XML documents.
+> Subcommands: `canonic`, `edit`, `elements`, `escape`, `unescape`, `format`, `list`, `pyx`, `depyx`, `select`, `transform`, or `validate`.
```

Comment:
How does the user learn the available subcommands? "xml --help" ?
Note that my text _does_ inform the user that there are subcommands _and_ provides a list.

Reply:
> XMLStarlet Toolkit: Query, edit, check, convert and transform XML documents.
> This command also has documentation about its subcommands, e.g. `xml validate`.
> Subcommands: `canonic`, `edit`, `elements`, `escape`, `unescape`, `format`, `list`, `pyx`, `depyx`, `select`, `transform`, or `validate`.
> More information: <http://xmlstar.sourceforge.net/docs.php>.

Reply:
When only writing `Subcommands: ` the user probably doesn't know that we also have tldr pages for them. Also, I think listing every single subcomand doesn't really look very nice.

Reply:
I wasn't trying to look nice. I'm trying to help a novice command-line user that doesn't know anything about the command and wants quick and easy answers. Listing the subcommands helps that user, saving them the step of finding the subcommands in the xml --help command.

Reply:
I don't know which I personally prefer, for our standard, marchersimons suggestion is better.

Also huge command libraries don't have this, like git.

Reply:
Also some clients support autosuggestions so you can just type `tldr xml ` and press Tab to get all subcommands.

Reply:
Revised:
-----
> XMLStarlet Toolkit: Query, edit, check, convert and transform XML documents.
> This command also has documentation about its subcommands, e.g. `xml validate`.
> More information: <http://xmlstar.sourceforge.net/docs.php>.

- Display general help, including the list of subcommands:

`xml --help`

Reply:
I like the new description, but we always keep the `--help` example at the bottom. Subcommands are important, but not neccessarily the first thing you want to know about a command.

Reply:
The `xml` command does nothing (except --help) without a subcommand, so the list of subcommands IS actually "necessarily the first thing you want to know about" the xml command. Without a list of subcommands, the next thing that a user must type after "tldr xml" is "xml --help". That's why I moved it to the top.

I appreciate that you have guidelines and conventions like "we always keep the --help example at the bottom" but I feel that you sometimes lose sight of the purpose of your program, which is to help the user. I feel that you should give priority to providing that help and communicating clearly and efficiently over nuances of syntax, semantics,  and formatting. Your audience is human and humans are not compilers.

Reply:
> The `xml` command does nothing (except --help) without a subcommand, so the list of subcommands IS actually "necessarily the first thing you want to know about" the xml command.

Oh, in this case let's keep it at the top.

> I appreciate that you have guidelines and conventions like "we always keep the --help example at the bottom" but I feel that you sometimes lose sight of the purpose of your program, which is to help the user.

That's something I also worry about. But to me it's just not possible to think through every page I review, since I can't bring up that time. Sticking to guidelines makes the review process much faster. I know the quality goes down a bit, but what would be the alternative?

Reply:
More written guidance, both for the authors and the reviewers, including meta-guidance on how strictly or loosely to enforce guidelines. 
--> More reviewers. Raise the required review count if you're worried about review quality.
Prioritize pragmatism over perfection. Don't stress over little stuff. :-)